src/Compilers/CSharp/Portable/Syntax/InternalSyntax/BinaryExpressionSyntax.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax
+{
+    internal partial class BinaryExpressionSyntax
```

Comment:
This adds protection for BinaryExpressionSyntax only -- should all of WriteTo be changed (probably via the generator) to use a non-recursive implementation?